PyRoute/TradeCodes.py

```diff
@@ -45,25 +80,21 @@ def __init__(self, initial_codes):
         self.sophont_list = []
         homeworlds_found = []
 
-        self.sophont_list = [code for code in self.codes if re.match(r"[\w\']{4}(\d|W)", code, re.U)]
-
-        for homeworld in re.findall(r"[Di]*\([^)]+\)\d?", initial_codes, re.U):
-            full_name = re.sub(r'\(([^)]+)\)\d?', r'\1', homeworld)
-            homeworlds_found.append(homeworld)
-            match = re.match(r'\w{,2}\(([^)]{,4})[^)]*\)(\d|W)?', homeworld)
-            if match is None:
-                self.logger.error("Unable to process %s", initial_codes)
-                sys.exit(1)
-            if full_name.startswith("Di"):
-                sophont = "{code: <4}{pop}".format(code=match.group(1), pop='X')
-            else:
-                sophont = "{code: <4}{pop}".format(code=match.group(1), pop=match.group(2) if match.group(2) else 'W')
+        self.sophont_list = [code for code in self.codes if re.match(r"[A-Za-z\'!]{1}[\w\'!]{2,4}(\d|W|\?)", code, re.U)]
```

Comment:
Can we move the sophont and homeworld list processing code to it's own procedure. This is just complicated enough I would take it out and be able to write tests around it.

Reply:
Brute-force hoisted out